connector/datadogconnector/go.mod

```
@@ -0,0 +1,88 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector
+
+go 1.20
```

We are still using Go 1.19, this will be changed when #25116 is addressed

```suggestion
go 1.19
```